### ISSUE TITLE
ScrollablePositionedList constructors are missing asserts for non-nullable parameters

### ISSUE THREAD

**Issue by marcglasberg** [google/flutter.widgets#101]
`ScrollablePositionedList` constructors are missing asserts for non-nullable parameters (it doesn't matter if they have default values, they still must assert):

```dart
assert(initialScrollIndex != null),
assert(initialAlignment != null),
assert(scrollDirection != null),
assert(reverse != null),
assert(addSemanticIndexes != null),
assert(addAutomaticKeepAlives != null),
assert(addRepaintBoundaries != null),
```

**Comment by jamesderlin** [comment]
Would be nice to do.  OTOH, this also will not be necessary when NNBD is enabled.

**Comment by tarobins** [comment]
NNBD, we need you.

**Comment by jamesderlin** [comment]
Should be fixed with 6cdc1ff,
